webapp/src/main/java/com/box/l10n/mojito/service/security/user/UserRepository.java

```diff
@@ -56,4 +56,17 @@ and lower(concat(u.givenName, ' ', u.surname)) like lower(concat('%', :search, '
       """)
   Page<User> findByUsernameOrName(
       @Param("username") String username, @Param("search") String search, Pageable pageable);
+
+  @Query(
+      value =
+          """
+    SELECT u FROM  User u
+      WHERE COALESCE(u.username, '') <> ''
+        AND LOWER(u.username) LIKE CONCAT(LOWER(:servicePrefix), '%')
+        AND LOWER(:serviceName) LIKE CONCAT(LOWER(u.username), '%')
+        AND u.enabled IS TRUE
+    """)
+  @EntityGraph(value = "User.legacy", type = EntityGraphType.FETCH)
+  Optional<List<User>> findService(
```

Comment:
Change method name to `findByServiceNameOrPrefix`

Reply:
Done